includes/functions.php

```diff
@@ -1432,17 +1432,29 @@ function cache_mac_oui()
     if ($lock->get()) {
         echo 'Caching Mac OUI' . PHP_EOL;
         try {
-            $mac_oui_url = 'https://macaddress.io/database/macaddress.io-db.json';
+            $mac_oui_url = 'https://raw.githubusercontent.com/wireshark/wireshark/master/manuf';
             echo '  -> Downloading ...' . PHP_EOL;
             $get = Requests::get($mac_oui_url, [], ['proxy' => get_proxy()]);
-            echo '  -> Processing ...' . PHP_EOL;
-            $json_data = $get->body;
-            foreach (explode("\n", $json_data) as $json_line) {
-                $entry = json_decode($json_line);
-                if ($entry && $entry->{'assignmentBlockSize'} == 'MA-L') {
-                    $oui = strtolower(str_replace(':', '', $entry->{'oui'}));
+            echo '  -> Processing CSV ...' . PHP_EOL;
+            $csv_data = $get->body;
+            foreach (explode("\n", $csv_data) as $csv_line) {
+                unset($oui);
+                $entry = str_getcsv($csv_line, "\t");
+                if (is_array($entry) && count($entry) >= 3 && strlen($entry[0]) == 8) {
+                    // We have a standard OUI xx:xx:xx
+                    $oui = strtolower(str_replace(':', '', $entry[0]));
+                } elseif (is_array($entry) && count($entry) >= 3 && strlen($entry[0]) == 20) {
+                    // We have a smaller range (xx:xx:xx:X or xx:xx:xx:xx:X)
+                    if (substr($entry[0], -2) == '28') {
+                        $oui = strtolower(str_replace(':', '', substr($entry[0], 0, 10)));
+                    } elseif (substr($entry[0], -2) == '36') {
+                        $oui = strtolower(str_replace(':', '', substr($entry[0], 0, 13)));
```

Comment:
you run `strtolower(str_replace(':', ''` 3 times, it can be 1 in the end

Reply:
right. Will do.

Reply:
done